Livening & Pairing up projectiles for fun and aesthetics, especially Blasters

ravenshining opened the pull request:
**Balance**

This makes a few tweaks to direct fire projectile weapons.  The goal being to make them less unwieldy, and to be more competitive with lasers.

It does **not** alter range or overall DPS, only how they fly, the balance of shield:hull damage, and adds some hit force.

- Blasters and slower projectiles have had their speed multiplied by 1.5 while preserving range, while projectiles in the 15-20 range have seen a more modest boost.
- Blasters have had their **damage doubled** and **reload halved**
- Modified Blaster hull/shield damage balance has been skewed slightly to encourage disabling by pirates.
- Blasters have been given a *tiny* amount of **hit force** to aid in escape prevention, in line with other non-laser weaponry.
- Twin blasters have had their mass/space **reduced** by 1 tonne
- For visual appeal and as a counterbalance, all multiple-gun blasters and the plasma turret have been set to fire two (or four in the case of the quad) shot bursts at the base reload, instead of firing individual shots at a higher rate.

## Testing Done
Shooting up pirates and stuff.

## Things That Could be Done but May (or may not) Happen in a Later PR:
 - [ ] Double-sounds
 - [x] Human guns
 - [x] Hai guns
 - [ ] Gegno acumen 7,7 40 ball 12 66 Astut 12 29 cholera 13 38 & 16 37 acuit 19 50
 - [ ] Coal bomby 9,2 97 ion hail 17 28 ion rain 20 20
 - [ ] Bun Locust 15 27 Thorax 18 100
 - [ ] Incipias Plasma 15 43
 - [ ] Remnant Thrasher 15 24
 - [ ] Korath Repeater 18 29
 - [ ] Ka'Het Annihilator 20 32
 - [ ] Gegno Irate pairing ?
 - [ ] Successor drag, bimodal pairing ?
 - [ ] Avigi speck pairing ?
 - [ ] Bunrodea mandible pairing ?
 - [ ] Inhibitor turret pairing ?

![Screenshot_20250111_022020](https://github.com/user-attachments/assets/4074c2aa-4bb3-4446-bf55-d6e3fd06b356)
![Screenshot_20250111_022202](https://github.com/user-attachments/assets/3d9df5df-a8bc-472b-a145-dd17879cc810)
![Screenshot_20250111_024004](https://github.com/user-attachments/assets/65b199df-4bc8-45b1-b00f-543074c96adf)
![Screenshot_20250111_024504](https://github.com/user-attachments/assets/63ec1d63-4ea7-48fd-ad47-fa0bff3fa14f)

TheGiraffe3 commented:
So essentially, blasters are four times as fast, _and_ are better at disabling/destroying?
That could get... interesting.

ravenshining commented:
Update (untested):
- pulse cannon & gatling bullets also sped up by x2
- railgun slugs sped up by x1.25

Deliberately leaving other things alone given special abilities or lack of interaction

ravenshining commented:
> So essentially, blasters are four times as fast, _and_ are better at disabling/destroying? That could get... interesting.

No, the blasts only move twice as fast.  They are better at bringing down shields, and more likely to disable, while being *worse* at damaging hull.

To clarify:  Individual shots move twice as fast, but they don't fly for as long:  range remains the same
Damage is doubled, but fire rate is halved: DPS on average remains the same
For multiples, they fire in bursts but less often: DPS on average remains the same

What this does is make blaster shots reach their targets quicker, adds space between them, and makes them more shield-oriented.

Zitchas commented:
Hmmm.... I just spent some time trying them out. My first impression is that they feel like a particle canon. I felt pressure to time my shots, and whenever I just held down the fire key it felt like I was missing most of the time. Tried tangling with an Argosy (I was flying a scout, so not really a great matchup anyway...) and I got decimated by the fast stream of fire from its turret.

Blasters just didn't feel like blasters. I don't know, might just be my bias. I'll do another test run of them. 

I like the change to the shield/hull balance, but the firing all at once I'm not really liking. 


As an aside, could you update this please? (or would you mind if I hit the update button?)

ravenshining commented:
Hm, I always play with automatic fire on.   Aiming and timing shots is my usually my gunner's job heh

If the grouping changes the feel too much, I can certainly dial that back.  The intention there was to make it possible for a small ship to get missed between bursts, rather than simply outrunning the shots as they did before.

Have you tried the other weapons?  They've all been sped up proportionately.

There aren't any conflicts so I don't see any point to updating but if you want to have at it.

ravenshining commented:
I had a thought about feeling too much like a particle cannon, and I realised there was a bunch of facing inaccuracy that... didn't make sense and was only there because copy and paste.  I've removed that, and shots are more tightly grouped now, not spreading out randomly.

Maybe that has the opposite effect from what you were looking for regarding fire rate, but should feel more like blasters again?

Also I tested plasma turret and fixed its submunition.

Zitchas commented:
Sounds interesting. I'm not sure when I will be able to test, but I will.

ravenshining commented:
Given the above feedback and extensive feedback on both Swizzle6 and ES Revamp, I've dialled back these changes significantly.

Instead of doubling blaster speed, they've only been increased to 15-16.  This value was chosen as it is one standard deviation above the speed of all the game's secondary, explosive, homing, or otherwise special weapons.

Weapons with a speed faster than a blaster but lower than 20 have been shifted to the 16-20 range, while weapons higher than 20 have not been and will not be affected.  20 is the speed of the Drak antimatter cannon, which is a good benchmark for dodgeability.

I've cut the hit force, and removed the energy balance change on standard blasters.  Modified blasters still have a tweaked ratio, but it's only to 2.33 -- I kept their hull damage the same as regular blasters while increasing the shield damage to keep DPS constant.  The idea here is that since modified blasters are primarily meant to be used by pirates, they would be the ones wanting to modify weapons for better capture rates.  Hopefully, this will result in more disabled merchants rather than more disabled pirates!

I'd like to extend the scope of this PR to more than just humans and Hai so it's all balanced; maybe I'll tackle that tomorrow.

Zitchas commented:
One thing that comes to mind regarding the shots being paired... Sometimes I like it when they are paired, sometimes I prefer the look of streamed. And the big thing being: I'm not consistent. I really wish I could change how they fired. Sometimes I think it'd be cool to have all the same-type guns on my ship all locked in sync and firing in unison. Other times I prefer to see a consistent stream of projectiles rather than bursts. 

Tactically speaking, if you want to lock something down (aka prevent them from fleeing to hyperspace), then stream of hits is better. On the other hand, if you want to deal enough damage to temporarily knock out the shields and do some more lasting damage to the hull, then paired is better because the damage arrives in larger spikes.

I'm not asking you to program that capability (although if you could, that'd be awesome...), it's just a thought that occured to me reading through this.

edit: just thinking about it, sometimes I'd even like to slow down my faster guns so that *all* my weapons fire in sync...

Zitchas commented:
Another thought. Not something that you need to solve, but just more thoughts and ideas:

a) Re: Blasters. It's OK for a gun to be bad. Being the cheapest gun around for penny-pinchers that just want *a* gun/turret but don't want to spend anything more than the absolute minimum in order to have a gun in the slot is a perfectly valid, and in fact, going by real world examples, actually probably one of the most sought after attribute to have. Admitedly, that is sought after by less reputable rental security agencies, cheapskate commercial organizations, militaries that don't care about individual capabilities, thugs going for appearance over capability, and people who are mainly interested in intimidating unarmed people (or who just want something to keep said people from having an edge on them. They don't want to fight, they just want to have a gun so that the thugs with guns don't think they are unarmed and might pick a softer target). None of which typically encompasses the player. Just saying. 

b) While I think there's a solid reason for leaving regular blasters alone and being terrible (maybe drop their price a bit...) This doesn't apply to all the blaster variants. Double blasters, mod blasters, etc. These are obviously people trying to accomplish something with them, otherwise they'd be perfectly happy just with a generic cheapo single blaster. So makes sense that these will be better in some way. Whether it is tweaked performance, faster rates of fire, whatever, they're trying it.

ravenshining commented:
My reasoning with the paired shots is that it gives a visual indication of what the ship has equipped.  Otherwise, two blasters look no different from double blasters.

Re: bad... Weak is one thing, but frustratingly unwieldy is another.  It seems quite silly to me to have direct-fire, non-laser weapons that are both weak *and* slower than most missiles.  And, I haven't made the blaster any stronger - it still does the same DPS.  Just more responsive, as I made projectiles more responsive in general.

If it were to make sense for any blasters to be objectively better than another, I would say that belongs to the Modified Blaster series.  It's both in the name, and they at least have a counterbalance of being poorly distributed.

Zitchas commented:
Updated as of 2025-02-21

Zitchas commented:
I have fond memories of the light blasters from EVN. I don't recall exactly how they compare to regular blasters, but I have the vague recollection that they were faster moving shots, faster firing, smaller, and *maybe* tighter firing pattern, although slightly lower damage that was completely outweighed by the improvements to usability. I always considered the regular blasters of EVN to be kind of like the Reaction thrusters: Cheapest weapon available, and thus the default manufacturer-provided thing; but replacing them with literally anything was a step up.

Zitchas commented:
I would like to express my deep appreciation for all the work you have done and your willingness to experiment and try out new things. You have been great to work with, and I hope to have opportunities to do so in the future.

For now, however, upstream's leaders have demonstrated one more time that they fear the actions of anyone else having some perceived power no matter that three years of proven track record are evidence against it being used in that way; and have gone a step forward in demonstrating that when given that power, they immediately do what they accuse other people of wanting to do. Hypocrisy and a demonstration that making decisions based on fear seldom work out.

This is not a healthy environment, so I will not, unfortunately, be having the time to dedicate to giving your work here the attention it deserves. If the situation changes, then I hope this can be re-opened and we can restart our work on it. I think there's some good ideas here.
